### ISSUE TITLE
Allow looping of the queue.

### ISSUE THREAD

**Issue by TheKrush** [angelobreuer/Lavalink4NET#110]
**Describe the solution you'd like**
SkipAsync can be modified in such a way to loop the queue instead of just a single track, I've done this by overriding SkipAsync in my own player.

**Describe alternatives you've considered**
I made my own player and overrode the SkipAsync functionality. But I can see where other people would also like to be able to loop the queue.

**Comment by TheKrush** [comment]
Here's my crude first pass of this and it seems to work for all cases I can think of.

```
        public override Task SkipAsync(int count = 1)
        {
            // we want to override this so we can change the looping functionality

            // no tracks to skip
            if (count <= 0)
            {
                return Task.CompletedTask;
            }

            EnsureNotDestroyed();
            EnsureConnected();

            // the looping option is enabled, repeat current track, does not matter how often we skip
            if (IsLooping && CurrentTrack != null)
            {
                // queue ths track at the end
                PlayAsync(CurrentTrack, enqueue: true);
            }
            
            // tracks are enqueued
            if (!Queue.IsEmpty)
            {
                LavalinkTrack? track = null;

                while (count-- > 0)
                {
                    if (IsLooping && track != null)
                    {
                        // add us to the end if we're looping
                        PlayAsync(track, enqueue: true);
                    }

                    // no more tracks in queue
                    if (Queue.Count < 1)
                    {
                        StopAsync(disconnect: _disconnectOnStop);
                        break;
                    }

                    // dequeue track
                    track = Queue.Dequeue();
                }

                // a track to play was found, dequeue and play
                return PlayAsync(track!, false);
            }
            // no tracks queued, stop player and disconnect if specified
            else
            {
                StopAsync(disconnect: _disconnectOnStop);
            }

            return Task.CompletedTask;
        }
```

**Comment by angelobreuer** [comment]
Sounds good! Would you like to file a PR?

I think it would be good to create a new property on the LavalinkPlayer, to support both modes.

```csharp
public enum PlayerLoopMode
{
    None,
    Track,
    Queue,
}
```

```diff
+ public PlayerLoopMode LoopMode { get; set; }
- public bool IsLooping { get; set; }
+ public bool IsLooping
+ {
+     get => LoopMode is not PlayerLoopMode.None;
+     set => LoopMode = value ? PlayerLoopMode.Track : PlayerLoopMode.None;
+ }
```

**Comment by TheKrush** [comment]
Yeah I can do that let me sync the repo, I've been working off NuGet.

**Comment by OneYellowLemon** [comment]
@TheKrush :wave: Are you planning to make a PR for this?

**Comment by TheKrush** [comment]
sorry about that it's up now

**Comment by angelobreuer** [comment]
Thank you, the changes will be included in preview 14

**Comment by angelobreuer** [comment]
Closed by #113
